rust/noosphere/src/wasm/fs.rs

```diff
@@ -0,0 +1,143 @@
+use crate::{
+    platform::{PlatformKeyMaterial, PlatformStorage},
+    wasm::SphereFile,
+};
+use js_sys::{Array, Function};
+use noosphere_fs::SphereFs as SphereFsImpl;
+use tokio_stream::StreamExt;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+/// A `SphereFs` is a view over the data of a `Sphere` that enables the user to
+/// think of sphere content in terms of file system semantics. You can think of
+/// the file system as having a non-hierarchical namespace, where each name
+/// points to a particular file. You can read or write to any name using strings
+/// or raw bytes.
+pub struct SphereFs {
+    #[wasm_bindgen(skip)]
+    pub inner: SphereFsImpl<PlatformStorage, PlatformKeyMaterial>,
+}
+
+#[wasm_bindgen]
+impl SphereFs {
+    #[wasm_bindgen]
+    /// Read a `SphereFile` from the sphere given a name in the namespace
+    pub async fn read(&self, slug: String) -> Result<Option<SphereFile>, String> {
+        let file = self
+            .inner
+            .read(&slug)
+            .await
+            .map_err(|error| format!("{:?}", error))?;
+
+        Ok(file.map(|file| SphereFile {
+            inner: file.boxed(),
+        }))
+    }
+
+    #[wasm_bindgen(js_name = "writeString")]
+    /// Write content to a name in the sphere's namespace as a UTF-8 encoded
+    /// string
```

Comment:
Does wasm bindgen handle WTF-16 => UTF-8 conversions?

Reply:
I believe so, yes. JS `string` to Rust `String` conversion is handled in the codegen.